jaxrs/src/main/java/com/cerner/beadledom/jaxrs/exceptionmapping/WebApplicationExceptionMapper.java

```diff
@@ -46,14 +46,16 @@ public Response toResponse(WebApplicationException exception) {
       logger.error("An unhandled exception was thrown.", exception);
     }
 
+    JsonError errorBody = response != null && response.getEntity() instanceof JsonError ? response
```

Comment:
Is this a fix for an existing bug or did something with the upgrade lead to this being necessary?

Reply:
Yeah, from resteasy 3.10+, the ExceptionHandler class of resteasy was updated to look for available exception mappers for a class and its super class. In one of my local services, I am throwing a BadRequestException with a JsonError entity which is a subClass of the WebApplicationException so this mapper catches it. So I have updated this mapper to honor any existing JsonError entity that could already be set by the consumer.

Reply:
7701ab8f I have refactored this to read a little better